src/op_mode/igmp-proxy.py

```diff
@@ -194,48 +146,73 @@ def do_mr_mfc():
                 'oifs'  : []
             }
 
-            iif = int( line.split()[2] )
+            iif = int(line.split()[2])
             if not ((iif == -1) or (iif == 65535)):
                 data['pkts']  = line.split()[3]
-                data['bytes'] = byte_string( line.split()[4] )
+                data['bytes'] = bytes_to_human(int(line.split()[4]))
                 data['wrong'] = line.split()[5]
 
                 # convert index to real interface name
                 data['iif']  = vif[iif]['interface']
 
                 # convert each output interface index to a real interface name
                 for oif in line.split()[6:]:
-                    idx = int( oif.split(':')[0] )
-                    data['oifs'].append( vif[idx]['interface'] )
+                    idx = int(oif.split(':')[0])
+                    data['oifs'].append(vif[idx]['interface'])
 
             result['data'].append(data)
 
     return result
 
-if __name__ == '__main__':
-    args = parser.parse_args()
-
-    # Do nothing if service is not configured
-    c = vyos.config.Config()
-    if not c.exists_effective('protocols igmp-proxy'):
-        print("IGMP proxy is not configured")
-        sys.exit(0)
-
-    if args.interface:
-        data = do_mr_vif()
-        if data:
-            tmpl = jinja2.Template(vif_out_tmpl)
-            print(tmpl.render(data))
-
-        sys.exit(0)
-    elif args.mfc:
-        data = do_mr_mfc()
-        if data:
-            tmpl = jinja2.Template(mfc_out_tmpl)
-            print(tmpl.render(data))
-
-        sys.exit(0)
-    else:
-        parser.print_help()
-        sys.exit(1)
 
+# Output template for "show ip multicast interface" command
+#
+# Example:
+# Interface  BytesIn      PktsIn       BytesOut     PktsOut      Local
+# eth0       0.0 B        0            0.0 B        0            xxx.xxx.xxx.65
+# eth1       0.0 B        0            0.0 B        0            xxx.xxx.xx.201
+# eth0.3     0.0 B        0            0.0 B        0            xxx.xxx.x.7
+# tun1       0.0 B        0            0.0 B        0            xxx.xxx.xxx.2
+def show_interface(raw: bool):
+    vif_out_template = """{%- for r in data -%}
+{{ "%-10s"|format(r.interface) }} {{ "%-12s"|format(r.bytes_in) }} {{ "%-12s"|format(r.pkts_in) }} {{ "%-12s"|format(r.bytes_out) }} {{ "%-12s"|format(r.pkts_out) }} {{ "%-15s"|format(r.loc) }}
+{% endfor %}"""
+    if data := _process_mr_vif():
+        if raw:
+            return json.loads(json.dumps(data))
+        else:
+            return jinja2.Template(vif_out_template).render(data)
```

Comment:
If I remember correctly the idea was to use `Tabulate` here over Jinja2 if possible to get nice looking tables.

Reply:
I wasn't sure whether I should hack on that part but I guess it's a good opportunity.